### DISCUSSION TITLE
NEED HELP: How to collapse tabs of the same type into a "tree"

### DISCUSSION THREAD

**Discussion by susieswallows** [piroor/treestyletab#3627] [Q&A] [answered]
I love the app by the way, I'd pretty neat and solves SOOOOO many of my headaches each time I use firefox/tor. I am asking this time about TOR btw. Can I collapse tabs of the same type into a "tree" without using the side bookmarks bar?  Is there a way I can collapse all the different family tabs in their respective trees from the top where the tabs usually are? Please don't tell me I'm stupid and blind and just can't see the option. 🤦

**Comment by SHHSSH** [comment]
Susie you shouldn't use addons necessarily if you are using TOR for privacy reasons.

**Reply by susieswallows** [reply]
Oh? Does it implement security vulnerabilities doing so? So, does this mean I shouldn't be using any addons for Tor whatsoever?

**Reply by SHHSSH** [reply]
> Oh? Does it implement security vulnerabilities doing so? So, does this mean I shouldn't be using any addons for Tor whatsoever?

Integrated addons that they provide with the TOR browser are all that you should be using, HTTPSEverywhere and NoScript, thankfully as TOR user perhaps you've garnered an understanding of the NoScript interface and I would suggest using that on the clearnet with clearnet browsers, it will provide you an understanding of certain aspects of the internet and will benefit you for sometime in the maelstrom of total confusion as to wtf this thing is being blocked and what not. 

^ This was a bit of a spiel on my experience's behalf, I am curious though as to how you came to be using TOR for a general usecase? Was it recommended to you by someone or had you come to use it by your own volition?

**Reply by susieswallows** [reply]
I've come to use it under my own volition, apparently it's the safest way to browse without my ISP knowing what activities I'd like to do. Been using Tor for that purpose for years though so I have a general understanding but I'm still learning everyday as I use it. Thankyou for all your help. Thumbed up!

**Comment by piroor** [comment, marked as answer]
TST doesn't have such a feature and there is no plan to do that due to the project policy. Thus you may need to implement a helper addon to do that - TST provides APIs for other addons to control its tree structure.
https://github.com/piroor/treestyletab/wiki/API-for-other-addons
